pkg/devfile/parser/configurables.go

```diff
@@ -0,0 +1,371 @@
+	metadata := d.Data.GetMetadata()
+	d.Data.SetMetadata(name, metadata.Version)
+	return d.WriteYamlDevfile()
+}
+
+// AddEnvVars adds environment variables to all the components in a devfile
+func (d DevfileObj) AddEnvVars(otherList []v1.EnvVar) error {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			component.Container.Env = Merge(component.Container.Env, otherList)
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+// RemoveEnvVars removes the environment variables which have the keys from all the components in a devfile
+func (d DevfileObj) RemoveEnvVars(keys []string) (err error) {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			component.Container.Env, err = RemoveEnvVarsFromList(component.Container.Env, keys)
+			if err != nil {
+				return err
+			}
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) setPorts(ports ...string) error {
+	components := d.Data.GetComponents()
+	endpoints, err := portsToEndpoints(ports...)
+	if err != nil {
+		return err
+	}
+	for _, component := range components {
+		if component.Container != nil {
+			component.Container.Endpoints = addEndpoints(component.Container.Endpoints, endpoints)
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) removePorts() error {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			component.Container.Endpoints = []v1.Endpoint{}
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) hasPorts() bool {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			if len(component.Container.Endpoints) > 0 {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (d DevfileObj) setMemory(memory string) error {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			component.Container.MemoryLimit = memory
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+func (d DevfileObj) GetMemory() string {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			if component.Container.MemoryLimit != "" {
+				return component.Container.MemoryLimit
+			}
+		}
+
+	}
+	return ""
+}
+
+func (d DevfileObj) GetMetadataName() string {
+	return d.Data.GetMetadata().Name
+}
+
+// FormatDevfileSupportedParameters outputs supported parameters and their description
+func FormatDevfileSupportedParameters() (result string) {
+	for _, v := range GetDevfileSupportedParameters() {
+		result = result + " " + v + " - " + supportedDevfileParameterDescriptions[v] + "\n"
```

Comment:
Consider using `fmt.Sprintf` here and a `[]String`, e.g.
```go
result = []String{"Available Parameters for Devfile Components:"}
for _, v := range GetDevfileSupportedParameters() {
  result = append(result, fmt.Sprintf("%s - %s", v, supportedDevfileParameterDescriptions[v]))
}
return strings.Join(result, "\n")
```

Reply:
Good catch! This function is for odo, and should be removed from the generic parser repo. 
`GetDevfileSupportedParameters()` is enough for the library